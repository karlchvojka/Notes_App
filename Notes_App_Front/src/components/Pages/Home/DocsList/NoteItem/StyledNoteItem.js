import styled from 'styled-components'
import {
  cyberFont,
  darkGrey,
  lightBlue,
} from 'src/css_vars.js'

const StyledNoteItem = styled.li`
  background-color: ${darkGrey};
  margin-bottom:20px;
  padding:10px;
  -webkit-box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);
  box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);

  .itemHeader {
    display:flex;
    justify-content: space-between;

    .itemHeadLeft {
      a:link, a:visited {
        color: ${lightBlue};
        display:block;
        font-family: ${cyberFont};
        font-size:25px;
        font-weight:700;
        margin-bottom:10px;
        text-transform: uppercase;
        text-decoration: none;
      }
      .ptitle {
        font-weight: bold;
        text-transform: uppercase;
      }
      .noteCat {
        font-size:15px;
        margin-bottom: 0px;
      }
      .noteDate {
        font-size:12px;
      }
    }
    .itemHeadRight {

      button {
        background-color: ${darkGrey};
        border: 0px;
        color: ${lightBlue};
        font-size:15px;
      }
      button:hover {
        cursor:pointer;
      }
      .editButton {
        background-color: ${darkGrey};
        color: ${lightBlue};
        font-size:15px;
        padding: 0px 5px;
        text-decoration: none;

      }
    }
  }
`

export default StyledNoteItem
