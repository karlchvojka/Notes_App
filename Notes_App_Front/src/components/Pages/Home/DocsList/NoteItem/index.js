import React, { useState, useEffect } from 'react';
import {
  BrowserRouter as Router,
  Link
} from "react-router-dom";
import dayjs from 'dayjs';
import styled from 'styled-components';
import { headerFont, paraFont, cyberFont, darkGrey, headerGrey, darkBlue, midBlue, lightBlue, lightestBlue } from 'src/css_vars.js';
import CrudHelpers from 'src/helpers/CrudHelpers.js';

import { FaPen, FaTrash } from 'react-icons/fa';

const StyledLi = styled.li`
  background-color: ${darkGrey};
  margin-bottom:20px;
  padding:10px;
  -webkit-box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);
  box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);

  .itemHeader {
    display:flex;
    justify-content: space-between;

    .itemHeadLeft {
      a:link, a:visited {
        color: ${lightBlue};
        display:block;
        font-family: ${cyberFont};
        font-size:25px;
        font-weight:700;
        margin-bottom:10px;
        text-transform: uppercase;
        text-decoration: none;
      }
      .ptitle {
        font-weight: bold;
        text-transform: uppercase;
      }
      .noteCat {
        font-size:15px;
        margin-bottom: 0px;
      }
      .noteDate {
        font-size:12px;
      }
    }
    .itemHeadRight {

      button {
        background-color:${darkGrey};
        border: 0px;
        color: ${lightBlue};
        font-size:15px;
      }
      button:hover {
        cursor:pointer;
      }
      .editButton {
        background-color:${darkGrey};
        color: ${lightBlue};
        font-size:15px;
        padding: 0px 5px;
        text-decoration: none;

      }
    }
  }
`;

const NoteItem = (props) => {
  const { note, notes, setNotes } = props;

  const handleDeleteClick = (e) => {
    CrudHelpers.deleteNote(e, note._id);
    setNotes(notes.filter(({ _id: i }) => note._id !== i));
  };

  return (
    <StyledLi>
      <div className="itemHeader">
        <div className="itemHeadLeft">
          <Link to={`/notes/${note._id}`}>{note.title}</Link>
          <p className="noteCat">
            <span className="ptitle">Category: </span>
            {note.category}
          </p>
          <p className="noteDate">
            <span className="ptitle">Posted: </span>
            {dayjs(`${note.date}`).format('DD/MM/YYYY')}
          </p>
        </div>
        <div className="itemHeadRight">
          <Link className="editButton" to={`/notes/${note._id}/edit`}><FaPen /></Link>
          <button className="deleteButton" onClick={e => handleDeleteClick(e)} type="button"><FaTrash /></button>
        </div>
      </div>
    </StyledLi>
  );
};

export default NoteItem;
