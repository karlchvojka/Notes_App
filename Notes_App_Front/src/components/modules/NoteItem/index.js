import React from "react"
import "./index.scss"

import CrudHelpers from "../../../helpers/CrudHelpers.js"

const NoteItem = (props) => {
  const { note, setNotes, notes } = props;

  const handleClick = (e) => {
    CrudHelpers.deleteNote(e, note._id)
    setNotes(notes.filter(({ _id: i }) => note._id !== i))
  }
  
  return (
    <li
      key={props.note._id}
    >
      {props.note.title}
      <button onClick={e => handleClick(e)}>X</button>
    </li>
  )
}

export default NoteItem
