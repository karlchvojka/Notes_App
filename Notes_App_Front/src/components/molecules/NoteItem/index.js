import React, { useState, useEffect } from 'react';
import './index.scss';
import {
  Editor,
  EditorState,
  convertFromRaw,
} from 'draft-js';
import CrudHelpers from '../../../helpers/CrudHelpers.js';

const NoteItem = (props) => {
  const { note, notes, setNotes } = props;
  const [editorState, setEditorState] = useState(EditorState.createEmpty());
  const [editorProps, setEditorProps] = useState({
    readOnly: true,
  });

  const loadContent = () => {
    const content = convertFromRaw(JSON.parse(note.content));
    setEditorState(EditorState.createWithContent(content));
  };

  useEffect(() => {
    loadContent();
  }, []);

  const handleClick = (e) => {
    CrudHelpers.deleteNote(e, note._id);
    setNotes(notes.filter(({ _id: i }) => note._id !== i));
  };

  return (
    <li>
      {note.title}
      <Editor editorState={editorState} readOnly={editorProps.readOnly} />
      <button className="deleteButton" onClick={e => handleClick(e)} type="button">X</button>
    </li>
  );
};

export default NoteItem;
