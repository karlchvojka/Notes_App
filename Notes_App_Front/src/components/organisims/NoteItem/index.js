import React, { useState, useEffect } from 'react';
import './index.scss';
import {
  BrowserRouter as Router,
  Link
} from "react-router-dom";
import dayjs from 'dayjs';
import CrudHelpers from '../../../helpers/CrudHelpers.js';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPen } from '@fortawesome/free-solid-svg-icons'

const NoteItem = (props) => {
  const { note, notes, setNotes } = props;

  const handleDeleteClick = (e) => {
    CrudHelpers.deleteNote(e, note._id);
    setNotes(notes.filter(({ _id: i }) => note._id !== i));
  };

  return (
    <li className="noteItem">
      <div className="itemHeader">
        <div className="itemHeadLeft">
          <Link to={`/notes/${note._id}`}>{note.title}</Link>
          <p className="noteCat">
            <span className="ptitle">Category: </span>
            {note.category}
          </p>
          <p className="noteDate">
            <span className="ptitle">Posted: </span>
            {dayjs(`${note.date}`).format('DD/MM/YYYY')}
          </p>
        </div>
        <div className="itemHeadRight">
          <Link className="editButton" to={`/notes/${note._id}/edit`}><FontAwesomeIcon icon={faPen} /></Link>
          <button className="deleteButton" onClick={e => handleDeleteClick(e)} type="button">X</button>
        </div>
      </div>
    </li>
  );
};

export default NoteItem;
