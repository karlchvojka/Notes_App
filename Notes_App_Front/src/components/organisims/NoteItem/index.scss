@import '../../../../styles/_reset.scss';
@import '../../../../styles/variables.scss';
@import '../../../../styles/fonts.scss';

.noteItem {
  background-color: $darkGrey;
  margin-bottom:20px;
  padding:10px;
  -webkit-box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);
  box-shadow: 0px 0px 7px 5px rgba(48,191,242,0.36);

  .itemHeader {
    display:flex;
    justify-content: space-between;

    .itemHeadLeft {
      a:link, a:visited {
        color: $lightBlue;
        display:block;
        font-family: $cyberFont;
        font-size:25px;
        font-weight:700;
        margin-bottom:10px;
        text-transform: uppercase;
        text-decoration: none;
      }
      .ptitle {
        font-weight: bold;
        text-transform: uppercase;
      }
      .noteCat {
        font-size:15px;
        margin-bottom: 0px;
      }
      .noteDate {
        font-size:12px;
      }
    }
    .itemHeadRight {

      button {
        background-color:$darkGrey;
        border: 1px solid $lightBlue;
        color: $lightBlue;
      }
      .editButton {
        background-color:$darkGrey;
        border: 1px solid $lightBlue;
        color: $lightBlue;
        font-family: $cyberFont;
        padding: 0px 5px;
        text-decoration: none;

      }
    }
  }
}
