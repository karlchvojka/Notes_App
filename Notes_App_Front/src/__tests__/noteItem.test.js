import React from "react";

import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';

import NoteItem from "../components/modules/NoteItem/";

afterEach(cleanup);

describe('when the app renders', () => {
  const note = {
    _id: 1,
    title: "Test note",
    content: "The test note content",
    date:"2020-02-02",
    category: "test"
  };

  const notes = [
    {
      _id: 0,
      title: "Test note",
      content: "The test note content",
      date:"2020-02-02",
      category: "test"
    },
    {
      _id: 1,
      title: "Test note1",
      content: "The test note1 content",
      date:"2020-02-01",
      category: "test"
    },
    {
      _id: 2,
      title: "Test note2",
      content: "The test note2 content",
      date:"2020-02-02",
      category: "test"
    },
  ]

  const wrapper = renderer.create(
    <NoteItem
      key={note._id}
      note={note}
      notes={notes}
    />
  );

  it("renders without crashing", () => {
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls the handleClick function on delete button", async () => {
    const handleClick = jest.fn();
    const setNotes = jest.fn();

    const { getByText } = render(<NoteItem
        key={note._id}
        note={note}
        notes={notes}
        setNotes={setNotes}
      />);

    fireEvent.click(getByText('X'));
    await expect(handleClick).toHaveBeenCalled();
  })


});
