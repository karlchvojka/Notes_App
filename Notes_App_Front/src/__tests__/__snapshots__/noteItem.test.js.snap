// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the app renders renders without crashing 1`] = `
<li>
  Test note
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    X
  </button>
</li>
`;
